controllers/operatorconfig_controller.go

```diff
@@ -54,51 +78,315 @@ type OperatorConfigReconciler struct {
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
 func (r *OperatorConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	log := log.FromContext(ctx)
+	// Fetch the OperatorConfig operatorConfig
+	operatorConfig := &cranev1alpha1.OperatorConfig{}
+	if err := r.Get(ctx, types.NamespacedName{
+		Namespace: InstallNamespace,
+		Name:      OwnerConfigName,
+	}, operatorConfig); err != nil {
+		// Error reading the object - requeue the request.
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			log.Info("Desired OperatorConfig resource not found")
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "Failed to get OperatorConfig")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// Add Finalizer for this CR
+	if !controllerutil.ContainsFinalizer(operatorConfig, Finalizer) {
+		controllerutil.AddFinalizer(operatorConfig, Finalizer)
+		err := r.Update(ctx, operatorConfig)
+		if err != nil {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if operatorConfig.DeletionTimestamp != nil {
+		// clean up
+		err := r.cleanUpResources(ctx)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		controllerutil.RemoveFinalizer(operatorConfig, Finalizer)
+		err = r.Update(ctx, operatorConfig)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		log.Info("Clean up successful")
+		return ctrl.Result{}, nil
+	}
+
+	for _, f := range filepaths {
+		err := r.createResourcesFromFile(f, ctx, log, operatorConfig)
+		if err != nil {
+			log.Error(err, "Error creating resources")
+			return ctrl.Result{Requeue: true}, nil
+		}
+	}
 
-	err := createClusterTasks(ctx, log)
+	log.Info("All needed resources are present")
```

Comment:
a poll amongst reviewers: should we remove this log entry?

My logging philosophy is to log on edge changes and not on a steady state. steady-state logging entries tend to blog up the operator logs. But this one is succinct so I will not oppose if folks prefer to keep it

Reply:
It should go.